configuration/pih/petl/jobs/hiv/viral_load/source.sql

```diff
@@ -0,0 +1,122 @@
+#### This query returns a row per encounter (VL construct per encounter)
+
+SET sql_safe_updates = 0;
+
+DROP TEMPORARY TABLE IF EXISTS temp_hiv_construct_encounters;
+DROP TEMPORARY TABLE IF EXISTS temp_vl_index_asc;
+DROP TEMPORARY TABLE IF EXISTS temp_vl_index_desc;
+
+### hiv constructs table
+CREATE TEMPORARY TABLE temp_hiv_construct_encounters
+(
+    patient_id                      INT,
+    encounter_id                    INT,
+    vl_sample_taken_date            DATETIME,
+    date_created                    DATETIME,
+    vl_sample_taken_date_estimated  VARCHAR(11),
+    vl_result_date                  DATE,
+    vl_test_outcome                 VARCHAR(255),
+    vl_result_detectable            VARCHAR(255),
+    viral_load                      INT,
+    vl_type                         VARCHAR(50)
+);
+
+-- patient and encounter IDs
+INSERT INTO temp_hiv_construct_encounters (patient_id, encounter_id)
+SELECT person_id, encounter_id FROM obs WHERE voided = 0 AND concept_id = CONCEPT_FROM_MAPPING("PIH", "HIV viral load construct");
+
+-- specimen collection date
+UPDATE temp_hiv_construct_encounters tvl JOIN encounter e ON tvl.encounter_id = e.encounter_id
+SET	vl_sample_taken_date = e.encounter_datetime;
+
+-- date encounter was created
+UPDATE temp_hiv_construct_encounters tvl JOIN encounter e ON tvl.encounter_id = e.encounter_id
+SET	tvl.date_created = e.date_created;
+
+## Delete test patients
+DELETE FROM temp_hiv_construct_encounters WHERE
+patient_id IN (
+               SELECT
+                      a.person_id
+                      FROM person_attribute a
+                      INNER JOIN person_attribute_type t ON a.person_attribute_type_id = t.person_attribute_type_id
+                      AND a.value = 'true' AND t.name = 'Test Patient'
+               );
```

Comment:
Tangetially, I don't think I did this for the hiv_patients table, we should correct that there as well.

Reply:
Done